addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3847,6 +3851,30 @@ export class OdooEditor extends EventTarget {
                             setSelection(codeElement.firstChild, 0);
                         }
                     }
+                } else if ((shouldCreateNumberList || shouldCreateBulletList) &&
+                    !closestElement(selection.anchorNode, 'li')
+                ) {
+                    this.historyStep();
+                    const node = selection.anchorNode;
+                    const listFormatRegex = /^[\-*]\s|^\b(?:[1aA])[.)]\s*/;
```

Comment:
Where did star `*` at the end came from here do you want to replace all the space after the number or alphabet?
What if I have `1.[]       abc` and I press space, all those space after the cursor will be gone...
I suggest this `/^(?:[1aA][.)]|[-*])\s/` but make sure you test it extensively.